src/pytest_cases/case_funcs.py

```diff
@@ -366,3 +366,30 @@ def is_case_function(f,                       # type: Any
         except:
             # GH#287: safe fallback
             return False
+
+
+def with_case_tags(*tags):
+    """Attach `tags` to all cases defined in the decorated class."""
+    def _decorator(cls):
+        if is_case_function(cls):
+            raise ValueError(
+                'Cannot use with_case_tags on a case '
+                'function. Use the @case decorator instead.'
+                )
+        if not is_case_class(cls):
+            raise ValueError('with_case_tags can only be applied to classes '
+                             'defining a collection of cases.')
+        for case_name in dir(cls):
+            case_ = getattr(cls, case_name)
+            if not is_case_function(case_):  # Not a case
+                continue
+            try:
+                case_info = getattr(case_, CASE_FIELD)
+            except AttributeError:
+                # Not explicitly decorated with @case. Do so now.
+                case_ = case(case_)
+                case_info = getattr(case_, CASE_FIELD)
```

Comment:
You will need to test this. If I remember correctly, by chance the `@case` decorator does not create a wrapper but simply modifies the decorated function. So you can probably write

```suggestion
                _ = case(case_)  # the case decorator directly modifies `_case`, no need to catch the output
                case_info = getattr(case_, CASE_FIELD)
```

Reply:
I see. Yes, you're absolutely right. I guess my intention was to also support possible future changes where the `@case` decorator returns a different callable , but clearly forgot to `setattr(cls, case_name, case_)` after decoration. Do you think we can guarantee that `@case` will always return the same callable? If not, I'd rather do the rebinding explicitly just for future-proofing.

Reply:
I guess that the test will be your guarantee here. In the future if `@case` returns a wrapper, then the tag will not be present and your test will fail. So that code will not be released.
My suggestion is therefore to not use `setattr` for now. But you can insert this in a comment line :

`# In the future if `@case` modifies the method, we will have to `setattr(cls, case_name, case_mod)``

Reply:
6b6259ba8223715b1cded58a0dbe2862a9e29716